runtime/autoload/provider/pythonx.vim

```diff
@@ -47,32 +47,27 @@ function! provider#pythonx#Require(host) abort
 endfunction
 
 function! provider#pythonx#Detect(major_ver) abort
-  let host_var = (a:major_ver == 2) ?
-        \ 'g:python_host_prog' : 'g:python3_host_prog'
-  let skip_var = (a:major_ver == 2) ?
-        \ 'g:python_host_skip_check' : 'g:python3_host_skip_check'
-  let skip = exists(skip_var) ? {skip_var} : 0
-  if exists(host_var)
-    " Disable auto detection.
-    let [result, err] = s:check_interpreter({host_var}, a:major_ver, skip)
-    if result
-      return [{host_var}, err]
+  if a:major_ver == 2
+    if exists('g:python_host_prog')
+      return [g:python_host_prog, '']
+    else
+      let progs = ['python2', 'python2.7', 'python2.6', 'python']
+    endif
+  else
+    if exists('g:python3_host_prog')
+      return [g:python3_host_prog, '']
```

Comment:
> Or perhaps CheckHealth could do that.

That. :> If someone is willed to specify an absolute path, he hopefully makes sure that it's a correct one anyway. Otherwise `:CheckHealth` will complain:

```
  - INFO: Using: g:python_host_prog = "snowball"
  - ERROR: "snowball" was not found.
  - INFO: Executable: Not found
```

But if you want, I change it to `if exists() && executable()`. It's probably just an additional `stat(2)`.